src/routes/transactions/mappers/transactions-history.mapper.ts

```diff
@@ -58,71 +44,62 @@ export class TransactionsHistoryMapper {
     if (transactionsDomain.length == 0) {
       return [];
     }
-    const previousTransaction = this.getPreviousItem(
-      offset,
-      transactionsDomain,
-    );
-    let prevPageTimestamp = 0;
-    if (previousTransaction !== null) {
-      prevPageTimestamp = this.getDayFromTransactionDate(
-        previousTransaction,
-        timezoneOffset,
-      ).getTime();
+    const previousTransaction = await (async (): Promise<
+      TransactionItem | undefined
+    > => {
+      const prevDomainTransaction = this.getPreviousItem(
+        offset,
+        transactionsDomain,
+      );
+      if (!prevDomainTransaction) {
+        return;
+      }
+      // We map in order to filter last list item against it
+      const mappedPreviousTransaction = await this.mapTransaction(
+        prevDomainTransaction,
+        chainId,
+        safe,
+        onlyTrusted,
+      );
       // Remove first transaction that was requested to get previous day timestamp
       transactionsDomain = transactionsDomain.slice(1);
-    }
 
-    const transactionsDomainGroups = this.groupByDay(
-      transactionsDomain,
-      timezoneOffset,
-    );
-
-    const transactionList = await Promise.all(
-      transactionsDomainGroups.map(async (transactionGroup) => {
-        const items: (TransactionItem | DateLabel)[] = [];
-        const groupTransactions = (
-          await this.mapGroupTransactions(
-            transactionGroup,
-            chainId,
-            safe,
-            onlyTrusted,
-          )
-        )
-          .filter(<T>(x: T | undefined): x is T => x != null)
-          .flat();
+      return Array.isArray(mappedPreviousTransaction)
+        ? // All transfers should have same execution date but the last is "true" previous
+          mappedPreviousTransaction.at(-1)
+        : mappedPreviousTransaction;
+    })();
+
+    const transactions = await (async (): Promise<Array<TransactionItem>> => {
+      const mappedTransactions = await Promise.all(
+        transactionsDomain.map((transaction) => {
+          return this.mapTransaction(transaction, chainId, safe, onlyTrusted);
+        }),
+      );
+      return mappedTransactions
+        .filter(<T>(x: T): x is NonNullable<T> => x != null)
+        .flat();
+    })();
```

Comment:
I'm not against this implementation, but have you considering extracting these to private functions instead of using IIFEs? Do you see an advantage on using the latter?

Reply:
I originally only had an IIFE for retrieving the "previous" transaction to keep the `slice` in the main function as it can be somewhat confusing otherwise. I've moved it to it's own function and added a relevant comment in eada02d1a1f2eb2ea6b9dc2bf4ed5af165149105.

As for the `mappedTransaction` IIFE, it's unnecessary and I removed it.